error LNK2001: unresolved external symbol for absl

When building from the latest 1.14.0 source code, I encountered numerous LNK2001 errors. 

Here are my development environment details:

.NET version: 8.0.404
CMake version: 3.26.3
Visual Studio: 2022

I followed these steps:

1. `dotnet publish Reinterop~ -o .`
2. Opened Unity.
3. `cmake -B build -S . -DCMAKE_BUILD_TYPE=Debug`
4. `cmake --build build -j14 --target install --config Debug`

After completing these steps, I received multiple LNK2001 errors. Some examples include:

Could you provide any insights into why this might be occurring?

Thank you in advance for your assistance.

`s2.lib(s2coords.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(r2rect.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(coder.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
CesiumGeospatial.lib(S2CellID.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s1interval.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s2cell_id.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s2latlng.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s2coords.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(r2rect.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(coder.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
CesiumGeospatial.lib(S2CellID.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s1interval.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s2cell_id.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s2latlng.cc.obj) : error LNK2001: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(s2coords.cc.obj) : error LNK2001: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > * __cdecl absl::lts_20240722::log_internal::MakeCheckOpString<__int64,__int64>(__int64,__int64,char const *)" (??$MakeCheckOpString@_J_J@log_internal@lts_20240722@absl@@YAPEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@_J0PEBD@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
s2.lib(r2rect.cc.obj) : error LNK2001: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > * __cdecl absl::lts_20240722::log_internal::MakeCheckOpString<__int64,__int64>(__int64,__int64,char const *)" (??$MakeCheckOpString@_J_J@log_internal@lts_20240722@absl@@YAPEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@_J0PEBD@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
CesiumGeospatial.lib(S2CellID.obj) : error LNK2001: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > * __cdecl absl::lts_20240722::log_internal::MakeCheckOpString<__int64,__int64>(__int64,__int64,char const *)" (??$MakeCheckOpString@_J_J@log_internal@lts_20240722@absl@@YAPEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@_J0PEBD@Z) [F:\project\ttttttttttttttttttttttttttttttttt\Packages\com.cesium.unity\native~\build\Runtime\CesiumForUnityNative-Runtime.vcxproj]
`

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,11 @@
 # Change Log
 
+## ? - ?
+
+##### Fixes :wrench:
+
+- Fixed a bug in the Abseil vcpkg overlay port that could cause linker errors on some systems.
+
 ## v1.14.0 - 2024-12-02
 
 ##### Additions :tada:

--- a/native~/vcpkg/ports/abseil/portfile.cmake
+++ b/native~/vcpkg/ports/abseil/portfile.cmake
@@ -27,16 +27,6 @@ endif()
 # Don't let Abseil clobber our CMAKE_MSVC_RUNTIME_LIBRARY choice.
 vcpkg_replace_string("${SOURCE_PATH}/CMakeLists.txt" "set(CMAKE_MSVC_RUNTIME_LIBRARY \"MultiThreaded$<$<CONFIG:Debug>:Debug>DLL\")" "#set(CMAKE_MSVC_RUNTIME_LIBRARY \"MultiThreaded$<$<CONFIG:Debug>:Debug>DLL\")")
 
-vcpkg_cmake_configure(
-    SOURCE_PATH "${SOURCE_PATH}"
-    DISABLE_PARALLEL_CONFIGURE
-    OPTIONS
-        -DABSL_PROPAGATE_CXX_STD=OFF
-        -DCMAKE_CXX_STANDARD=14
-        ${ABSL_USE_CXX17_OPTION}
-        ${ABSL_STATIC_RUNTIME_OPTION}
-)
-
 # Don't let our customized version of Abseil pose as the real thing.
 vcpkg_replace_string("${SOURCE_PATH}/absl/base/options.h" "ABSL_OPTION_INLINE_NAMESPACE_NAME lts_20240722" "ABSL_OPTION_INLINE_NAMESPACE_NAME lts_20240722_cesium_for_unreal")
 vcpkg_replace_string("${SOURCE_PATH}/absl/base/options.h" "ABSL_OPTION_USE_STD_ANY 2" "ABSL_OPTION_USE_STD_ANY 0")
@@ -51,6 +41,16 @@ vcpkg_replace_string("${SOURCE_PATH}/absl/base/config.h" "#define ABSL_LTS_RELEA
 # https://github.com/abseil/abseil-cpp/pull/1728
 vcpkg_replace_string("${SOURCE_PATH}/absl/time/time.h" "__cpp_impl_three_way_comparison" "__cpp_lib_three_way_comparison")
 vcpkg_replace_string("${SOURCE_PATH}/absl/strings/cord.h" "__cpp_impl_three_way_comparison" "__cpp_lib_three_way_comparison")
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    DISABLE_PARALLEL_CONFIGURE
+    OPTIONS
+        -DABSL_PROPAGATE_CXX_STD=OFF
+        -DCMAKE_CXX_STANDARD=14
+        ${ABSL_USE_CXX17_OPTION}
+        ${ABSL_STATIC_RUNTIME_OPTION}
+)
 
 vcpkg_cmake_install()
 vcpkg_cmake_config_fixup(PACKAGE_NAME absl CONFIG_PATH lib/cmake/absl)

